### PR TITLE
AiSD-10-mergeSort

### PR DESCRIPTION
**Pull request by mztezi**

### CODE REVIEW

File: AiSD-10-mergeSort.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    left = mergeSort(left);
+    right = mergeSort(right);
+
+    return merge(left, right);
+}
+
+function merge(left, right) {
+    let output = [];
+
+    while (left.length > 0 && right.length > 0) {
```
Comment (kdryzek) [inline AiSD-10-mergeSort.js]: to be optimized

File: AiSD-10-mergeSort.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        if (left[0] < right[0]) {
+            output = [...output, left[0]];
+            left.shift();
+        } else {
+            output = [...output, right[0]];
+            right.shift();
+        }
+    }
+
+    while (left.length > 0) {
```
Comment (kdryzek) [inline AiSD-10-mergeSort.js]: to be optimized

File: AiSD-10-mergeSort.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        output = [...output, left[0]];
+        left.shift();
+    }
+
+    while (right.length > 0) {
```
Comment (kdryzek) [inline AiSD-10-mergeSort.js]: to be optimized

File: AiSD-10-mergeSort.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+function mergeSort(arr) {
+    if (arr.length < 2) {
+        return arr;
+    }
+
+    let left = [];
+    let right = [];
+
+    for (let i = 0; i < arr.length; i++) {
```
Comment (kdryzek) [inline AiSD-10-mergeSort.js]: use proper array method instead

File: AiSD-10-mergeSort.js (hunk 1 above)
Comment (mztezi) [reply AiSD-10-mergeSort.js]: discussed, fixed

File: AiSD-10-mergeSort.js (hunk 2 above)
Comment (mztezi) [reply AiSD-10-mergeSort.js]: discussed, fixed

File: AiSD-10-mergeSort.js (hunk 3 above)
Comment (mztezi) [reply AiSD-10-mergeSort.js]: discussed, fixed

File: AiSD-10-mergeSort.js (hunk 4 above)
Comment (mztezi) [reply AiSD-10-mergeSort.js]: discussed, fixed

File: src/algorithms/mergeSort.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+function mergeSort(arr) {
+    if (arr.length < 2) {
+        return arr;
+    }
+
+    const middle = Math.floor(arr.length / 2);
+    let left = arr.slice(0, middle);
```
Comment (kdryzek) [inline src/algorithms/mergeSort.js]: const here and below

File: src/algorithms/mergeSort.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    let right = arr.slice(middle);
+
+    return merge(mergeSort(left), mergeSort(right));
+}
+
+function merge(left, right) {
+    let output = [];
+
+    while (left.length && right.length) {
+        left[0] < right[0] ? output = [...output, left.shift()] : output = [...output, right.shift()];
```
Comment (kdryzek) [inline src/algorithms/mergeSort.js]: nope; to discuss;
```javascript
output = ....
```
or better
```javascript 
output.push(....
```

File: src/algorithms/mergeSort.js (hunk 5 above)
Comment (mztezi) [reply src/algorithms/mergeSort.js]: fixed

File: src/algorithms/mergeSort.js (hunk 6 above)
Comment (mztezi) [reply src/algorithms/mergeSort.js]: discussed, ok

**Review by kdryzek — APPROVED** [review]

**Pull request merged**